src/main/java/com/google/jspecify/nullness/ConformanceTypeInformationPresenter.java

```diff
@@ -0,0 +1,107 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.jspecify.nullness;
+
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeFormatter;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.util.visualize.AbstractTypeInformationPresenter;
+import org.checkerframework.framework.util.visualize.TypeOccurrenceKind;
+
+public class ConformanceTypeInformationPresenter extends AbstractTypeInformationPresenter {
+
+  /**
+   * Constructs a presenter for the given factory.
+   *
+   * @param atypeFactory the AnnotatedTypeFactory for the current analysis
+   */
+  public ConformanceTypeInformationPresenter(AnnotatedTypeFactory atypeFactory) {
+    super(atypeFactory);
+  }
+
+  @Override
+  protected AnnotatedTypeFormatter createTypeFormatter() {
+    // Use the same type formatter as normal error messages. Look into whether a different format
+    // would be better here.
+    return atypeFactory.getAnnotatedTypeFormatter();
+  }
+
+  @Override
+  protected TypeInformationReporter createTypeInformationReporter(ClassTree tree) {
+    return new ConformanceTypeInformationReporter(tree);
+  }
+
+  class ConformanceTypeInformationReporter extends TypeInformationReporter {
+    ConformanceTypeInformationReporter(ClassTree tree) {
+      super(tree);
+    }
+
+    @Override
+    protected void reportTreeType(
+        Tree tree, AnnotatedTypeMirror type, TypeOccurrenceKind occurrenceKind) {
+
+      String direction = null;
+      String usage = null;
+
+      switch (tree.getKind()) {
+        case RETURN:
+          direction = "sinkType";
+          usage = "return";
+          break;
+
+        case METHOD_INVOCATION:
+          AnnotatedExecutableType calledType = (AnnotatedExecutableType) type;
+          List<? extends AnnotatedTypeMirror> params = calledType.getParameterTypes();
+          MethodInvocationTree mit = (MethodInvocationTree) tree;
+          List<? extends ExpressionTree> args = mit.getArguments();
+          assert params.size() == args.size();
+
+          for (int i = 0; i < params.size(); ++i) {
+            checker.reportError(
+                tree,
+                "sinkType",
+                typeFormatter.format(params.get(i)),
+                // TODO: build up something from method and parameter name?
```

Comment:
What I've done so far is `methodName#paramName`. See [_Basic.java_ line 42](https://github.com/jspecify/jspecify/blob/main/conformance-tests/src/assertions/java/org/jspecify/conformance/tests/Basic.java#L42).

Reply:
Done.